src/donationswap.py

```diff
@@ -681,25 +813,55 @@ def _get_match_score(self, offer_a, offer_b, db):
 		score = round(score, 4)
 		return score, reason
 
-	def _get_actual_amounts(self, my_offer, their_offer):
-		if self._currency.is_more_money(
+	def _get_actual_amounts(self, match, my_offer, their_offer, db):
+
+		# cache check
+		if (match.new_amount_suggested > 0 and match.old_amount_suggested > 0):
+			if (match.new_offer_id == my_offer.id and match.old_offer_id == their_offer.id):
+				return match.new_amount_suggested, match.old_amount_suggested
+			elif (match.old_offer_id == my_offer.id and match.new_offer_id == their_offer.id):
+				return match.old_amount_suggested, match.new_amount_suggested
+
+		currencyData = self._currency
+
+		try:
+			one_day_ago = datetime.datetime.utcnow() - datetime.timedelta(days=1)
+			if (match.created_ts < one_day_ago) :
+				url = 'http://data.fixer.io/api/{:04d}-{:02d}-{:02d}?access_key=%s' % (match.created_ts.year, match.created_ts.month, match.created_ts.day, self._secret)
```

Comment:
Historic Currency is just that - the currency rates at a given point in time (given day).
you are correct the intent is to settle on an amount and then keep using that amount to avoid "you've been matched to donate $534" -> "now that both people have accepted please donate the $499"

534 -> 499 -> not great
534 -> 535 -> acceptable
534 -> 13249 -> unacceptable (I'm not aware of this happening, but we don't have any record of the value assigned to each swap so I can't definitively say an exchange rate didn't crash after creating a match)